## Importing libraries

In [1]:
import pandas as pd
import numpy as np

## Load Data

In [2]:
dataset=pd.read_csv("titanicsurvival.csv")

In [3]:
dataset

Pclass     Sex   Age     Fare  Survived
0         3    male  22.0   7.2500         0
1         1  female  38.0  71.2833         1
2         3  female  26.0   7.9250         1
3         1  female  35.0  53.1000         1
4         3    male  35.0   8.0500         0
..      ...     ...   ...      ...       ...
886       2    male  27.0  13.0000         0
887       1  female  19.0  30.0000         1
888       3  female   NaN  23.4500         0
889       1    male  26.0  30.0000         1
890       3    male  32.0   7.7500         0

[891 rows x 5 columns]

- dataset variable set hold the complete information of dataset.

## Summarize data set

In [4]:
print(dataset.shape)
print(dataset.head(5))

(891, 5)
   Pclass     Sex   Age     Fare  Survived
0       3    male  22.0   7.2500         0
1       1  female  38.0  71.2833         1
2       3  female  26.0   7.9250         1
3       1  female  35.0  53.1000         1
4       3    male  35.0   8.0500         0


- Here we get the complete information of dataset 
- dataset(dataset.shape) - It shows 891 Columns and 5 Rows
- dataset(dataset.shape) - It shows top 5 results of the dataset

  ## Mappping Text data into Binary data

In [5]:
income_set=set(dataset["Sex"])
dataset["Sex"]=dataset["Sex"].map({"female":0, "male":1}).astype(int)
print(dataset.head)

<bound method NDFrame.head of      Pclass  Sex   Age     Fare  Survived
0         3    1  22.0   7.2500         0
1         1    0  38.0  71.2833         1
2         3    0  26.0   7.9250         1
3         1    0  35.0  53.1000         1
4         3    1  35.0   8.0500         0
..      ...  ...   ...      ...       ...
886       2    1  27.0  13.0000         0
887       1    0  19.0  30.0000         1
888       3    0   NaN  23.4500         0
889       1    1  26.0  30.0000         1
890       3    1  32.0   7.7500         0

[891 rows x 5 columns]>


- Here we Mappping Text data into Binary data

## Segregate the data

In [6]:
X=dataset.drop("Survived",axis="columns")

In [7]:
X

Pclass  Sex   Age     Fare
0         3    1  22.0   7.2500
1         1    0  38.0  71.2833
2         3    0  26.0   7.9250
3         1    0  35.0  53.1000
4         3    1  35.0   8.0500
..      ...  ...   ...      ...
886       2    1  27.0  13.0000
887       1    0  19.0  30.0000
888       3    0   NaN  23.4500
889       1    1  26.0  30.0000
890       3    1  32.0   7.7500

[891 rows x 4 columns]

- Here we got the values of X

In [8]:
Y=dataset.Survived

In [9]:
Y

0      0
1      1
2      1
3      1
4      0
      ..
886    0
887    1
888    0
889    1
890    0
Name: Survived, Length: 891, dtype: int64

- Here we got the values of Y

## FInding and removing null values from features X

In [10]:
X.columns[X.isna().any()]  # Finding null value at a time in all the four coloumns

Index(['Age'], dtype='object')

In [11]:
X.Age=X.Age.fillna(X.Age.mean())  # Filling null values in "Age" Coloumn

In [12]:
X

Pclass  Sex        Age     Fare
0         3    1  22.000000   7.2500
1         1    0  38.000000  71.2833
2         3    0  26.000000   7.9250
3         1    0  35.000000  53.1000
4         3    1  35.000000   8.0500
..      ...  ...        ...      ...
886       2    1  27.000000  13.0000
887       1    0  19.000000  30.0000
888       3    0  29.699118  23.4500
889       1    1  26.000000  30.0000
890       3    1  32.000000   7.7500

[891 rows x 4 columns]

- Here we fill tha null values in Age Column

## Test again to checka any null value

In [13]:
X.columns[X.isna().any()]

Index([], dtype='object')

In [14]:
X

Pclass  Sex        Age     Fare
0         3    1  22.000000   7.2500
1         1    0  38.000000  71.2833
2         3    0  26.000000   7.9250
3         1    0  35.000000  53.1000
4         3    1  35.000000   8.0500
..      ...  ...        ...      ...
886       2    1  27.000000  13.0000
887       1    0  19.000000  30.0000
888       3    0  29.699118  23.4500
889       1    1  26.000000  30.0000
890       3    1  32.000000   7.7500

[891 rows x 4 columns]

- Here we have tested again for null values

## Splitting data set in to trianing and Test

In [15]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,Y, test_size=0.25 , random_state=0)

- Here we are passing both input and output seperately using train_test_split

## Training

In [16]:
from sklearn.naive_bayes import GaussianNB
model=GaussianNB()
model.fit(X_train,y_train)

GaussianNB()

- Training with  GaussianNB

## Predicting wheather person survived or no

In [17]:
pclassNO=int(input("Enter persons Pclass no:"))
gender=int(input("Enter persons gender 0-female 1-mail(0 or 1):"))
age=int(input("Enter Persons Age:"))
fare=float(input("Enter Persons fare:"))
person=[[pclassNO,age,gender,fare]]
result=model.predict(person)
print(result)
if result == 1:
    print("Person might be survived")
else:
    ("Person might not be survived")

Enter persons Pclass no:61
Enter persons gender 0-female 1-mail(0 or 1):0
Enter Persons Age:35
Enter Persons fare:2.5
[1]
Person might be survived


- Here we Predicting wheather person survived or no

## Prediction for all Test Data

In [18]:
y_pred=model.predict(X_test)
print(np.column_stack((y_pred,y_test)))

[[0 0]
 [0 0]
 [0 0]
 [1 1]
 [1 1]
 [0 1]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [0 0]
 [1 1]
 [0 0]
 [1 1]
 [1 1]
 [1 0]
 [0 0]
 [0 0]
 [0 0]
 [0 1]
 [0 0]
 [1 1]
 [0 0]
 [0 0]
 [1 0]
 [1 1]
 [0 0]
 [1 1]
 [1 1]
 [1 0]
 [0 0]
 [1 1]
 [0 0]
 [0 1]
 [0 0]
 [0 1]
 [0 0]
 [0 0]
 [0 0]
 [0 0]
 [1 1]
 [0 0]
 [0 0]
 [0 0]
 [1 1]
 [0 0]
 [0 0]
 [1 1]
 [0 0]
 [1 0]
 [0 1]
 [0 1]
 [1 1]
 [0 0]
 [0 1]
 [0 0]
 [0 0]
 [1 0]
 [0 0]
 [0 1]
 [0 0]
 [1 0]
 [1 1]
 [0 0]
 [1 1]
 [0 0]
 [1 1]
 [0 0]
 [1 1]
 [1 1]
 [1 1]
 [0 1]
 [1 0]
 [0 0]
 [0 0]
 [1 1]
 [1 0]
 [0 0]
 [0 0]
 [0 0]
 [0 0]
 [0 1]
 [1 0]
 [0 0]
 [0 0]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [1 0]
 [0 0]
 [0 0]
 [0 1]
 [1 1]
 [1 0]
 [0 0]
 [1 1]
 [0 0]
 [0 0]
 [1 1]
 [0 0]
 [0 0]
 [0 0]
 [0 0]
 [1 0]
 [1 1]
 [1 1]
 [1 0]
 [0 0]
 [1 1]
 [0 0]
 [1 1]
 [0 1]
 [1 0]
 [1 1]
 [1 1]
 [1 1]
 [1 1]
 [0 0]
 [1 1]
 [0 1]
 [0 0]
 [0 0]
 [0 0]
 [0 0]
 [0 0]
 [0 0]
 [0 1]
 [0 0]
 [0 0]
 [1 0]
 [0 0]
 [0 0]
 [1 0]
 [0 0]
 [0 0]
 [0 0]
 [1 0]
 [0 0]
 [0 0]
 [0 0]
 [1 1]
 [0 0]

## Conclusion - Accuracy of Model

In [19]:
from sklearn.metrics import accuracy_score
print("Accuracy of the Model:{0}%".format(accuracy_score(y_test,y_pred)*100))

Accuracy of the Model:77.57847533632287%


- Conclusion - Accuracy of Model is 77.57847533632287%